docs/vivisect/devguides/vstruct.rst

```diff
@@ -0,0 +1,104 @@
+            self.NumberOfSections     = v_uint16()
+            self.TimeDateStamp        = v_uint32()
+            self.PointerToSymbolTable = v_uint32()
+            self.NumberOfSymbols      = v_uint32()
+            self.SizeOfOptionalHeader = v_uint16()
+            self.Characteristics      = v_uint16()
+
+More Flexible Data
+==================
+
+Sometimes protocols aren't always so straightforward. Sometimes strings aren't always zero terminated, sometimes one field implies the existence (or non-existence) of another depending on the value.
+
+To help with this, vstruct provides an in-band callback mechanism for each of the properties on the vstruct-derived classes. As each field is populated into the vstruct instance, any callback matching the name of the field is run in the order that the callbacks were added.
+
+There are two ways to add parsing callbacks. The first (and simplest) is a method on your custom vstruct class that matches the form "pcb_<fieldname>".
+
+If you need more than one callback when a VStruct property is set, you'll need to resort to the `vsAddParseCallback` method on an `instance` of your vstruct class, which allows for an arbitrary number of callbacks to be defined for a property::
+
+    import vstruct
+    from vstruct.primitives import *
+    class Header(vstruct.VStruct):
+        def __init__(self):
+            vstruct.VStruct.__init__(self)
+            self.size = v_int32()
+            self.byts = v_bytes()
+
+        def pcb_size(self):
+            vs.vsSetField('size', olds * 2)
+
+    def sizemod(vs):
+        size = vs.vsGetField('size')
+        vs.vsGetField('byts').vsSetLength(size)
+
+    inst = Header()
+    inst.vsAddParseCallback('size', sizemod)
+    with open('mybytes.bin', 'rb') as fd:
+        inst.vsparse(fd.read())
+
+The above example demonstrates a relatively common use case of the custom callbacks, first parsing some field representing the size of the following fields, and then retrieving the bytes of those subsequent fields.
+
+As one final note, between the `pcb_<fieldname>` method and the `vsAddParseCallback` way of adding callbacks, the `pcb_<fieldname>` runs before any callbacks defined by `vsAddParseCallback`, and then the callbacks added by `vsAddParseCallback` are run in the order they were added.
+
+More Flexible VStructs
+======================
+
+VStruct was designed to be flexible when parsing binary structures. Some binary formats (such as DWARF) would be extraordinarily difficult to produce all the possible class definitions. So instead, what we can is instantiate an empty base VStruct class, and then dynamically add in properties::
+
+    import vstruct
+    import vstruct.defs.pe
+    import vstruct.primitives
+    vs = vstruct.VStruct()
+    vs.vsAddField('field_name', vstruct.primitives.v_uint8())
+    vs.vsAddField('second_field', vstruct.primitives.v_uint32())
+    vs.vsAddField('complex_field', vstruct.defs.pe.IMAGE_FILE_HEADER())
```

Comment:
again, i'm pretty sure you can just do:

```
vs = vstruct.VStruct()
vs.field_name = vstruct.primitives.v_uint8()
...
```

because it's such a cool and easy to use feature of VStruct, i think we want that spelled out (perhaps after your example here, so they get to see the two different ways).